### DISCUSSION TITLE
Problem saving the script for sending the email

### DISCUSSION THREAD

**Discussion by sawacs** [arendst/Tasmota#12146] [Q&A] [answered]
**_I noticed a problem with a saved script that doesn't start after rebooting_**
I have an ESP32 with a script that has been working very well for a while. ESP32 communicates with the energy meter via rs485-modbys-rtu. I wrote a few lines more of script to get an e-mail with the measurements or when ESP32 restarts. Emails come in, everything theoretical works. But after a power outage or software reset, the script is turned off. I attach it, save it, and after restart the situation repeats itself. So I did a comments the part of the script that sends email. Then the script writes itself and runs after reboot. So ... I wrote a simple test script that sends an email when esp32 boots up and it has this problem too .... (Console command "script 1" does't help too!)

**My build configuration :**

#ifndef USE_SCRIPT
#define USE_SCRIPT
#endif
#ifndef USE_SML_M
#define USE_SML_M
#endif
#ifdef USE_RULES
#undef USE_RULES
#endif
#define MAX_METERS 2
#define SML_BSIZ 64
#define USE_SML_M
#define SML_REPLACE_VARS
#define USE_SML_SCRIPT_CMD
#define USE_SCRIPT_WEB_DISPLAY
#define USE_SCRIPT_JSON_EXPORT
#define USE_SCRIPT_SUB_COMMAND
#define USE_SCRIPT_TASK
#define USE_SCRIPT_TIMER
#define USE_GOOGLE_CHARTS
#define SCRIPT_FULL_WEBPAGE
#define MAXSVARS 20
#define USE_SENDMAIL
#define USE_ESP32MAIL
#define USE_UFILESYS

#define EMAIL_SERVER smtp.gmail.com
#define EMAIL_PORT 465
#define EMAIL_USER "myusername"
#define EMAIL_PASSWORD "mypassword"
#define EMAIL_FROM "myusername"@gmail.com

**Test script:**

>D 
email1="sawacssc@gmail.com"

>B
=>sendmail [*:*:*:*:*:%Email1%:TEST] Restart odczytu licznika

**Comment by barbudor** [comment]
Script being disabled (or rules) is generally the result of protection against reboot loops.
So for some reasons sending email ay crash tasmota which restarts in loop until the protection kicks in and disable the script 

If you can have a serial terminal to show the logs that could confirm the idea.

If confirmed, I don't know right now what could be the reason. May be the script tries to send the email before wifi is stable and the solution could be to delay the email.

**Comment by gemu2015** [comment, marked as answer]
you may NOT send emails in section >B or >BS. the same is true e.g. for websend or publish. (and for rules too)
this is because on a reboot there is no net available and sadly this leads to crashing and rebooting.
you may check if you have a connection by testing wifi status (wifis)
